master/custom/schedulers.py

```diff
@@ -18,8 +22,16 @@ def addBuildsetForChanges(self, **kwargs):
         chdict = yield self.master.db.changes.getChange(changeid)
 
         builder_filter = chdict["properties"].get("builderfilter", None)
+        event = chdict["properties"].get("event", None)
         builder_names = kwargs.get("builderNames", self.builderNames)
         if builder_filter and builder_names:
+            # allow unstable builders only for comment-based trigger
+            if event != "issue_comment":
+                builder_names = [
+                    builder_name
+                    for builder_name in builder_names
+                    if builder_name in self.stable_builder_names
+                ]
```

Comment:
Would it be possible to create the 2 build name lists only once in the constructor, rather than having to do it at each GitHub event?

Reply:
not in the general case - you can see in line 26 the initial value of `builder_names` can come from the `kwargs` of the `addBuildsetForChanges` call, so we can't assume anything in the constructor.

I am not an expert on buildbot internals, but I don't actually see where `addBuildsetForChanges` *can* be called with a `builderNames` kwarg. if we only ever use `self.builderNames` in practice, we can definitely pre-construct the list in the constructor.

I can make this change now, or perhaps add logging so we can see whether it is ever called with a `builderNames` kwarg.

Reply:
Oh ok, I was wrong, ignore my comment.